articles/libraries/auth0-react/index.md

```diff
@@ -0,0 +1,187 @@
+  - index
+---
+
+<!-- markdownlint-disable MD041 -->
+
+# Auth0 React SDK
+
+
+The Auth0 React SDK is a JavaScript library for implementing authentication & authorization in React apps with Auth0. It provides a custom React hook and other Higher Order Components so you can secure React apps using best practices while writing less code.
+
+The Auth0 React SDK handles grant and protocol details, token expiration and renewal, as well as token storage and cacheing. Under the hood, it implements [Universal Login](/universal-login) and the [Authorization Code Grant Flow with PKCE](/api-auth/tutorials/authorization-code-grant-pkce).
+
+The library is [hosted on GitHub](https://github.com/auth0/auth0-react) and you can find the API documentation [here](https://auth0.github.io/auth0-react/).
+
+## Installation
+
+You have a few options for using auth0-react in your project:
+
+From the CDN:
+
+```html
+<script src="${auth0react_url}"></script>
+```
+
+Using [npm](https://npmjs.org):
+
+```sh
+npm install @auth0/auth0-react
+```
+
+Using [yarn](https://yarnpkg.com):
+
+```sh
+yarn add @auth0/auth0-react
+```
+
+## Getting Started
+
+First, you'll need to wrap your application in a single `Auth0Provider` component. This will provide the React Context to components that are placed inside your application.
+
+```jsx
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Auth0Provider } from '@auth0/auth0-react';
+import App from './App';
+
+ReactDOM.render(
+  <Auth0Provider
+    domain="${account.namespace}"
+    clientId="${account.clientId}"
+    redirectUri={window.location.origin}
+  >
+    <App />
+  </Auth0Provider>,
+  document.getElementById('app')
+);
+```
+
+Use the `useAuth0` hook in your components to access the React Context's authentication state (`isLoading`, `isAuthenticated` and `user`) and authentication methods (`loginWithRedirect` and `logout`).
+
+```jsx
+import React from 'react';
+import { useAuth0 } from '@auth0/auth0-react';
+
+function App() {
+  const {
+    isLoading,
+    isAuthenticated,
+    error,
+    user,
+    loginWithRedirect,
+    logout,
+  } = useAuth0();
+
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
+  if (error) {
+    return <div>Oops... {error.message}</div>;
+  }
+  if (!isAuthenticated) {
+    return <button onClick={loginWithRedirect}>Log in</button>;
+  }
+  if (isAuthenticated) {
+    return (
+      <div>
+        Hello {user.name} <button onClick={logout}>Log out</button>
+      </div>
+    );
+  }
+}
+
+export default App;
+```
+
+### Use with a Class Component
+
+Use the `withAuth0` higher order component to add the `auth0` property to Class components instead of using the hook.
+
+```jsx
+import React, { Component } from 'react';
+import { withAuth0 } from '@auth0/auth0-react';
+
+class Profile extends Component {
+  render() {
+    const { user } = this.props.auth0;
+    return <div>Hello {user.name}</div>;
+  }
+}
+
+export default withAuth0(Profile);
+```
```

Comment:
This would be a great, concise example of showing the user profile if you were to split that out from the login/profile/logout sample from earlier in the article, except with functions instead of classes.